twilight-cache-inmemory/src/builder.rs

```diff
@@ -1,21 +1,114 @@
+use std::fmt::Debug;
+use std::marker::PhantomData;
+
+use twilight_model::{
+    channel::{Channel, StageInstance},
+    guild::{GuildIntegration, Role},
+    user::{CurrentUser, User},
+};
+
+use crate::{
+    model, ChannelInterface, CurrentUserInterface, EmojiInterface, GuildIntegrationInterface,
+    GuildInterface, MemberInterface, MessageInterface, PresenceInterface, RoleInterface,
+    StageInstanceInterface, StickerInterface, UserInterface, VoiceStateInterface,
+};
+
 use super::{
     config::{Config, ResourceType},
     InMemoryCache,
 };
 
 /// Builder to configure and construct an [`InMemoryCache`].
-#[derive(Debug, Default)]
+#[allow(clippy::type_complexity)]
 #[must_use = "has no effect if not built"]
-pub struct InMemoryCacheBuilder(Config);
+pub struct InMemoryCacheBuilder<
+    CachedChannel: ChannelInterface = Channel,
```

Comment:
Opinion on shortening these to "C\<Type\>" instead of "Cached\<Type\>"?

Reply:
I don't really mind either way. "C<Type>" is more convenient to type, but probably less self-explanatory